### PR TITLE
Clean up build

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,10 +49,6 @@
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>mailer</artifactId>
         </dependency>
-        <dependency>
-            <groupId>commons-lang</groupId>
-            <artifactId>commons-lang</artifactId>
-        </dependency>
```
Comment (basil) [inline pom.xml:55]: This dependency is provided by Jenkins core, so it doesn't need to be bundled in the plugin.

File: pom.xml [hunk 2]
```diff
@@ -76,14 +72,13 @@
         <changelist>-SNAPSHOT</changelist>
         <jenkins.version>2.332.1</jenkins.version>
         <gitHubRepo>jenkinsci/${project.artifactId}</gitHubRepo>
-        <findbugs.failOnError>false</findbugs.failOnError>
```
Comment (basil) [inline pom.xml:79]: SpotBugs seems to be passing, so let us run it in local builds. CI already passes `failonError=false`.

File: pom.xml [hunk 3]
```diff
@@ -76,14 +72,13 @@
...
     </properties>
 
     <scm>
-        <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
```
Comment (basil) [inline pom.xml:83]: The `git://` URLs are deprecated in favor of HTTPS URLs.

File: pom.xml [hunk 4]
```diff
@@ -76,14 +72,13 @@
...
+        <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
         <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
         <url>https://github.com/${gitHubRepo}</url>
-        <tag>build-user-vars-plugin-1.9</tag>
+        <tag>${scmTag}</tag>
```
Comment (basil) [inline pom.xml:81]: Should be set to `scmTag` for proper incremental builds.

**Review by fabiodcasilva — APPROVED** [review]

**Pull request merged**